python/lsst/daf/io/table/sql.py

```diff
@@ -0,0 +1,938 @@
+# -*- coding: utf-8 -*-
+"""
+Collection of query wrappers / abstractions to both facilitate data
+retrieval and to reduce dependency on DB-specific API.
+"""
+
+from __future__ import print_function, division
+from builtins import str, bytes, zip, range
+
+import numpy as np
+import sqlalchemy
+import struct
+
+from lsst.afw import table as afw_table
+from sqlalchemy import select
+from sqlalchemy import Table, Column, Numeric
+from sqlalchemy.schema import MetaData, CreateTable
+from sqlalchemy.types import (SmallInteger, Integer, BigInteger,
+                              Float, String, Boolean, Text)
+from sqlalchemy.types import to_instance, TypeEngine
+from contextlib import contextmanager
+
+
+MAX_ARRAY_SIZE = 65535
+MAX_VARCHAR_SIZE = 512
+
+
+def is_dict_like(obj):
+    return hasattr(obj, '__getitem__') and hasattr(obj, 'keys')
+
+
+def is_list_like(obj):
+    return hasattr(obj, '__iter__') and not isinstance(obj, (str, bytes))
+
+
+class DatabaseError(IOError):
+    pass
+
+
+def read_sql_table(table_name, con, schema=None, column_dtypes=None,
+                   coerce_float=True, parse_dates=None, columns=None,
+                   chunksize=None):
+    """Read SQL database table into an afw table.
+
+    Given a table name and an SQLAlchemy connectable, returns an Table.
+    This function does not support DBAPI connections.
+
+    Parameters
+    ----------
+    table_name : string
+        Name of SQL table in database
+    con : SQLAlchemy connectable (or database string URI)
+    schema : string, default None
+        Name of SQL schema in database to query (if database flavor
+        supports this). If None, use default schema (default).
+    column_dtypes : dict, default: None
+        Dict of ``{column_name: dtype}`` where dtype is the target
+        dtype of the column. This is used to harmonize types and downcast
+        from doubles and longs.
+    coerce_float : boolean, default True
+        Attempt to convert values of non-string, non-numeric objects (like
+        decimal.Decimal) to floating point. Can result in loss of Precision.
+    parse_dates : list or dict, default: None
+        - List of column names to parse as dates
+        - Dict of ``{column_name: format string}`` where format string is
```

Comment:
here you use double backticks `\`\``, but on line 71 (around "chunksize") you only have a single backpack.